corehq/apps/sms/api.py

```diff
@@ -643,7 +643,7 @@ def get_inbound_phone_entry(msg):
             p = None
             if toggles.ONE_PHONE_NUMBER_MULTIPLE_CONTACTS.enabled(backend.domain):
                 running_session_info = XFormsSessionSynchronization.get_running_session_info_for_channel(
-                    SMSChannel(backend_id=msg.backend_id, phone_number=msg.phone_number)
+                    SMSChannel(backend_id=msg.backend_id, phone_number=apply_leniency(msg.phone_number))
```

Comment:
@Robert-Costello I was wondering if we shouldn't apply this at the point where we calculate the cache key which would cover any future usages as well:

https://github.com/dimagi/commcare-hq/blob/e89002988ff1decd0e840c4234ce4833b6a313bc/corehq/apps/smsforms/models.py#L428

and 

https://github.com/dimagi/commcare-hq/blob/e89002988ff1decd0e840c4234ce4833b6a313bc/corehq/apps/smsforms/models.py#L433

Reply:
Having done that I think we can remove it from here.